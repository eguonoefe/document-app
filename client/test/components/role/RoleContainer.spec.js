/* global expect jest test */
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {
  RoleContainer
} from '../../../components/role/RoleContainer';

configure({ adapter: new Adapter() });

describe('Role Container', () => {
  const props = {
    role: { title: 'Eguono', id: 3, content: 'john', access: 'role' },
    createRole: jest.fn(() => {
      return Promise.resolve();
    }),
    history: {
      push: jest.fn()
    },
    match: {
      url: ''
    },
  };
  const component = shallow(<RoleContainer {...props} />);
  beforeEach(() => {
    global.Materialize = { toast: jest.fn() };
  });
  describe('Container', () => {
    test('should match snapshot of document Container', () => {
      const tree = toJson(component);
      expect(tree).toMatchSnapshot();
    });
    test('onChange function should run when called', () => {
      component.instance().onChange(
        { target: { value: 'eguono', name: 'title' } });
      component.instance().onChange(
        { target: { value: 'efe', name: 'description' } });
      expect(component.state('title')).toEqual('eguono');
      expect(component.state('description')).toEqual('efe');
      const spy = jest.spyOn(component.instance(), 'onSubmit');
      component.instance().onSubmit();
      expect(spy).toHaveBeenCalled();
    });
  });
});
