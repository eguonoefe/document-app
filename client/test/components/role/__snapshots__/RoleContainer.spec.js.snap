// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Role Container Container should match snapshot of document Container 1`] = `
<RoleDisplay
  onChange={[Function]}
  onSubmit={[Function]}
  role={
    Object {
      "description": "",
      "title": "",
    }
  }
/>
`;
